server/handler/assignments.go

```diff
@@ -45,14 +42,30 @@ type assignmentRequest struct {
 }
 
 // SaveAssignment returns a function that saves the user answers as passed in the body request
-func SaveAssignment() RequestProcessFunc {
+func SaveAssignment(repo *repository.Assignments) RequestProcessFunc {
 	return func(r *http.Request) (*serializer.Response, error) {
-		requestedAssignmentID := chi.URLParam(r, "assignmentId")
-		assignmentID, err := strconv.Atoi(requestedAssignmentID)
+		assignmentID, err := urlParamInt(r, "assignmentId")
 		if err != nil {
-			return nil, serializer.NewHTTPError(
-				http.StatusBadRequest, fmt.Sprintf("wrong format in assignment ID sent; received %s", requestedAssignmentID),
-			)
+			return nil, err
+		}
+
+		assignment, err := repo.GetByID(assignmentID)
+		if err != nil {
+			return nil, err
+		}
+
+		if assignment == nil {
+			return nil, serializer.NewHTTPError(http.StatusNotFound, "assignment not found")
+		}
+
+		userID, err := service.GetUserID(r.Context())
+		if err != nil {
+			return nil, err
+		}
+
+		if userID != assignment.UserID {
+			return nil, serializer.NewHTTPError(http.StatusUnauthorized,
```

Comment:
1. `StatusForbitten` may be better here. It's a common approach to `logout` user on `StatusUnauthorized` when there is a separate frontend application. Because you usually return this error in auth middleware. (for example when token is outdated)
2. is it better to `Get(id, userID)`?

not a blocker though